docs/run_test/troubleshoot_failures.rst

```diff
@@ -0,0 +1,240 @@
+
+  -  **ASSIGNED**
+  
+     ASSIGNED tests are tests that are assigned to an environment, and
+     will start to run, if applicable, once the environment is
+     deployed/initialized. They suggest some environmental setting up is
+     going on.
+
+     ASSIGNED tests will end with FAILED if the environment fails to
+     deploy. Otherwise, they move forward to RUNNING. They will also move
+     backward to QUEUED if the environment is deployed and initialized
+     successfully.
+
+  -  **RUNNING**
+  
+     RUNNING tests are tests that are in test procedure.
+     RUNNING tests will end with one of the following final results.
+
+- **Final results**
+
+A final result shows information of a terminated test. It provides more
+valuable information than the intermediate result. It only appears in
+the end of a successful test run.
+
+  -  **FAILED**
+  
+     FAILED tests are tests that did not finish successfully and
+     terminated because of failures like ``LISA exceptions`` or
+     ``Assertion failure``. You can use them to trace where the problem
+     was and why the problem happened.
+
+  -  **PASSED**
+
+     PASSED tests are tests that passed, or at least partially passed,
+     with a special ``PASSException`` that warns there are minor errors in
+     the run but they do not affect the test result.
+
+  -  **SKIPPED**
+
+     SKIPPED tests are tests that did not start and would no longer run.
+     They suggest failure to meet some requirements in the environments
+     involved with the test.
+
+  -  **ATTEMPTED**
+
+     ATTEMPTED tests are a special category of FAILED tests because of
+     known issues, which are not likely to be fixed soon.
+
+- **Console Output**
+
+The results of a test run are displayed in the console and saved in log
+files generated by LISA.  The console will display a summary at the end
+of each run, containing the test suite and case name, test status and a
+message if applicable.  There will be a summary generated that tallies
+results of all tests.
+
+.. figure:: ../img/test_results_summary.png
+   :alt: test_results_summary
+
+Failures are categorized by similar messages.
+
+Log Files
+--------------------
+
+After a test run, the LISA log file will be generated. The log file can
+be found in the `runtime/log` directory that is generated after test
+runs.  Navigate subfolders until you find the log with a timestamp
+corresponding to the time of the test run.  Inside the log's timestamped
+folder, the contents are further split by environment and test case. The
+logs will show INFO and above levels by default.
+
+- **LOG FOLDER STRUCTURE** 
+
+  * **environment** folder, which contains logs split for the
+    environment.
+  * **tests** folder, which contains logs split for the test cases.
+  * **lisa.html** A formatted summary of test results. It can be viewed
+    by opening the file in a web browser.
+  * **lisa-<timestamp>.log** A full log of the test run. It contains all
```

Comment:
Please mention, if the run has only a few test cases, the full log is easy to read. But if it's run with concurrency, a split log is easier to read.

Reply:
Added this